packages/block-editor/src/components/link-control/index.js

```diff
@@ -351,14 +408,16 @@ function LinkControl( {
 							showTextControl={ showTextControl }
 							showSettings={ showSettings }
 							textInputRef={ textInputRef }
-							internalTextInputValue={ internalTextInputValue }
+							internalTextInputValue={
+								internalControlValue?.title
+							}
 							setInternalTextInputValue={
 								setInternalTextInputValue
 							}
 							handleSubmitWithEnter={ handleSubmitWithEnter }
-							value={ value }
+							value={ internalControlValue }
 							settings={ settings }
-							onChange={ onChange }
+							onChange={ setInternalSettingValue }
```

Comment:
This changes means that any changes to settings are stored in internal state and not submitted until the user confirms with "Apply".